src/main/java/bio/terra/app/controller/SnapshotsApiController.java

```diff
@@ -273,13 +277,16 @@ public ResponseEntity<PolicyResponse> addSnapshotPolicyMember(
       @PathVariable("id") UUID id,
       @PathVariable("policyName") String policyName,
       @Valid @RequestBody PolicyMemberRequest policyMember) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
     PolicyModel policy =
         iamService.addPolicyMember(
-            getAuthenticatedInfo(),
-            IamResourceType.DATASNAPSHOT,
-            id,
-            policyName,
-            policyMember.getEmail());
+            userReq, IamResourceType.DATASNAPSHOT, id, policyName, policyMember.getEmail());
+    journalService.journalUpdate(
```

Comment:
Moving these Journal entries to the the IamService makes sense.  There are relatively few callers.  

In the bigger picture,  I'm trying to avoid noise in the journal with entries that can get lost in a long list.  IMHO, journaling every stariway flight might be an extension for the future after considering the noise it might add.  For now, only specific flights with big themes are journaled.